File: library/src/main/java/com/library/management/service/interfaces/DashboardService.java
package com.library.management.service.interfaces;
import com.library.management.dto.response.FineHistoryResponse;
import com.library.management.dto.response.DueSoonBookResponse;
import com.library.management.dto.response.DashboardResponse;
import com.library.management.dto.response.BorrowedBookDetailResponse;
import java.util.List;
import com.library.management.dto.response.MostBorrowedBookResponse;
import com.library.management.dto.response.TopActiveStudentResponse;
import com.library.management.dto.response.MonthlyBorrowStatisticsResponse;
import com.library.management.dto.response.NeverBorrowedBookResponse;
import com.library.management.dto.response.StudentDashboardResponse;
import com.library.management.dto.response.ReportResponse;
import com.library.management.dto.response.BorrowedBookResponse;

public interface DashboardService {
    StudentDashboardResponse getStudentDashboard(Long userId);
    DashboardResponse getDashboardSummary();
    List<MostBorrowedBookResponse> getMostBorrowedBooks();
    List<TopActiveStudentResponse> getTopActiveStudents();
    List<MonthlyBorrowStatisticsResponse> getMonthlyBorrowStatistics();
    List<NeverBorrowedBookResponse> getNeverBorrowedBooks();
    List<ReportResponse> getReport();
    List<BorrowedBookResponse> getBorrowedBooks(Long userId);
    List<DueSoonBookResponse> getDueSoonBooks(Long userId);
    BorrowedBookDetailResponse getBorrowedBookDetails(Long issueId);
    List<BorrowedBookResponse> getBorrowHistory(Long userId);
    List<FineHistoryResponse> getFineHistory(Long userId);
}

File: library-frontend/src/services/dashboardService.js
import api from "../api/axiosConfig";
export const getDashboardSummary = async () => {

    const response = await api.get("/dashboard");

    return response.data;

};

export const getStudentDashboard = async (userId) => {

    const response = await api.get(`/dashboard/student/${userId}`);

    return response.data;

};

export const getMostBorrowedBooks = async () => {

    const response = await api.get("/dashboard/most-borrowed-books");

    return response.data;

};

export const getTopActiveStudents = async () => {

    const response = await api.get("/dashboard/top-active-students");

    return response.data;

};

export const getMonthlyBorrowStatistics = async () => {

    const response = await api.get("/dashboard/monthly-borrow-statistics");

    return response.data;

};

export const getNeverBorrowedBooks = async () => {

    const response = await api.get("/dashboard/never-borrowed-books");

    return response.data;

};
export const getReports = async () => {

    const response = await api.get("/dashboard/reports");

    return response.data;

};
export const getDueSoonBooks = async (userId) => {

    const response = await api.get(
        `/dashboard/student/${userId}/due-soon`
    );

    return response.data;

};
export const getBorrowedBooks = async (userId) => {

    const response = await api.get(
        `/dashboard/student/${userId}/borrowed-books`
    );

    return response.data;

};
export const getBorrowedBookDetails = async (issueId) => {

    const response = await api.get(
        `/dashboard/borrowed-book/${issueId}`
    );

    return response.data;

};
export const getBorrowHistory = async (userId) => {

    const response = await api.get(
        `/dashboard/student/${userId}/history`
    );

    return response.data;

};
export const getBorrowHistoryBookDetails = async (issueId) => {

    const response = await api.get(
        `/dashboard/borrowed-book/${issueId}`
    );

    return response.data;

};
export const getFineHistory = async (userId) => {

    const response = await api.get(

        `/dashboard/student/${userId}/fine-history`

    );

    return response.data;

};